readthedocs/core/resolver.py

```diff
@@ -122,26 +122,21 @@ def resolve_path(self, project, filename='', version_slug=None,
     def resolve_domain(self, project, private=None):
         # pylint: disable=unused-argument
         canonical_project = self._get_canonical_project(project)
-        domain = canonical_project.domains.filter(canonical=True).first()
-        if domain:
-            return domain.domain
+        domains = canonical_project.domains.filter(canonical=True)
+        if domains.exists():
+            return domains.first().domain
         elif self._use_subdomain():
             return self._get_project_subdomain(canonical_project)
         return getattr(settings, 'PRODUCTION_DOMAIN')
 
-    def resolve(self, project, protocol='http', filename='', private=None,
+    def resolve(self, project, protocol='http', filename='', private=None, domain=None,
                 **kwargs):
-        if private is None:
-            version_slug = kwargs.get('version_slug')
-            if version_slug is None:
-                version_slug = project.get_default_version()
-            private = self._get_private(project, version_slug)
```

Comment:
Believe this code (`if private`..) needs to continue to exist.

Reply:
It seemed to me that this part is not necessarily needed as the `private` parameter is passed to the `resolve_path` function which is doing this same thing before using `private`. This would be increasing the number of queries if `private` evaluates to `False`.